fe/lib/components/SalaryDetails/SalaryDetails.tsx

```diff
@@ -87,6 +122,23 @@ export const SalaryDetails = ({
     min: validateMinAmount(min, errors),
   };
 
+  if (
+    payTypesLoading ||
+    currenciesLoading ||
+    payTypesData === undefined ||
+    currenciesData === undefined
+  ) {
+    return <Loader />;
+  }
+
+  const payTypes = payTypesData.payTypes.reduce<Record<string, PayType>>(
+    (acc, data) => ({
+      ...acc,
+      [`${data.basisCode}.${data.intervalCode}`]: data,
+    }),
+    {},
+  );
```

Comment:
Is this pass necessary? I would expect the query response to be deduplicated and in a reasonable order.

Reply:
Do you mean to just use the original response? I mainly adjust the shape so that it was easier for https://github.com/seek-oss/wingman/pull/959/files/448d70dbd088ab8055c79f4e187aac0d3580b146#diff-de34c4e404c87e05543daa353d41e70903945fea108a2430232043cd03912cabR266

Reply:
Oh I see, I was only looking at the `Object.entries` usage below. Would it make sense to change this to a `payType = data.payTypes.find((p) => p.label === payTypeLabel)` and simplify that part to `data.payTypes.map(...)`?

Reply:
Did you mean something like https://github.com/seek-oss/wingman/pull/959/commits/564ebf85389d8d823aa392403f875353a246079b?

Reply:
Ooh I see, there's a bit of complexity as we need the codes and the labels in different places. I think I'm fine with either approach having seen that, thanks for looking into this 🙇